woocommerce/woocommerce-ios
# No discard confirmation when exiting product details

Closes: #13571


## Description


There was no discard confirmation if I exit the product details with unsaved changes. The changes were just discarded with no warning.

Changes are explained in the files changed but basically some delegate calls are changed and navigation checks.

One thing that is still problematic is the swipe back gesture when in first screen of product details, that looks to be broken in general, here is a ticket I created for it: #13651

https://github.com/user-attachments/assets/51aefbdd-4339-4d63-92e5-41aa590a92ce

## Testing information


1. Go to the **Products** tab.
2. **Create** a new product or **select** an existing product.
3. **Add** or **change** product details. Do it on the initial product screen and detailed screens like product price
4. Tap the back button to exit the product details and notice there is confirmation sheet to discard changes or cancel.

## Screenshots


![Simulator Screen Recording - iPhone 15 Pro - 2024-08-19 at 11 35 03](https://github.com/user-attachments/assets/5864fa9c-77b9-470f-be41-1998fbbad020)

![Simulator Screen Recording - iPad za test - 2024-08-19 at 11 36 36](https://github.com/user-attachments/assets/86e3dec8-3700-4e33-abf9-4d36322be835)

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
